### PR TITLE
[src] remove redundant _scheduler_stack_check

### PR DESCRIPTION
**Pull request by polarvid**
## 拉取/合并请求描述：(PR description)

[

#### 为什么提交这份PR (why to submit this PR)

remove the redundant definition of `_scheduler_stack_check`

#### 你的解决方案是什么 (what is your solution)

Add a common definition `rt_scheduler_stack_check`

#### 请提供验证的bsp和config (provide the config and bsp) 

- BSP:

- .config:

- action:

]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [ ] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [ ] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [ ] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [ ] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [ ] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 代码是高质量的 Code in this PR is of high quality
- [ ] 已经使用[formatting](https://github.com/mysterywolf/formatting) 等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

File: src/scheduler_comm.c [hunk 1]
```diff
@@ -215,3 +215,63 @@ rt_err_t rt_sched_thread_change_priority(struct rt_thread *thread, rt_uint8_t pr
...
+    }
+#endif /* RT_USING_HW_STACK_GUARD */
+#ifdef ARCH_CPU_STACK_GROWS_UPWARD
+#ifndef RT_USING_HW_STACK_GUARD
+    else if ((rt_ubase_t)thread->sp > ((rt_ubase_t)thread->stack_addr + thread->stack_size))
+#else
+    if ((rt_ubase_t)thread->sp > ((rt_ubase_t)thread->stack_addr + thread->stack_size))
+#endif
+    {
+        rt_kprintf("warning: %s stack is close to the top of stack address.\n",
```
Comment (mysterywolf) [inline src/scheduler_comm.c:261]:
```suggestion
        LOG_W("%s stack is close to the top of stack address",
```

Comment (mysterywolf) [reply src/scheduler_comm.c:261]: rt_kprintf仅在终端交互场景中使用，警告要是用LOG_x

**Review by mysterywolf — CHANGES_REQUESTED** [review]

**Comment by polarvid** [comment]
> @polarvid 这份PR是否要需要合并到5.1.0？

不是必要的呢。只是调整一下代码样式。

**Review by mysterywolf — APPROVED** [review]
如果要是样式改变的话  那5.1.0就合并了吧

**Review by Rbb666 — APPROVED** [review]

**Pull request merged**